packages/typescript-eslint/src/config-helper.ts

```diff
@@ -133,12 +152,16 @@ export function config(
         const name = [config.name, extension.name].filter(Boolean).join('__');
         return {
           ...extension,
-          ...(config.files && { files: config.files }),
-          ...(config.ignores && { ignores: config.ignores }),
+          ...Object.fromEntries(
+            Object.entries(config).filter(([key]) =>
+              ['files', 'ignores'].includes(key),
+            ),
+          ),
           ...(name && { name }),
         };
       }),
-      config,
+      // `config` could be any object, but we'll assume it's a `Config` object for TS purposes.
+      config as Config,
```

Comment:
Inside this function, `config` should be `unknown`, but outside this function, `config` should be `Config` even if it's really an invalid value